topology/m4/mixercontrol.m4

```diff
@@ -28,6 +28,47 @@ define(`CONTROLMIXER_OPS',
 `		put STR($4)'
 `	}')
 
+dnl KCONTROL_2CHANNELS(reg)
+define(`KCONTROL_2CHANNELS',
+`KCONTROL_CHANNEL(FL, $1, 0),'
+`KCONTROL_CHANNEL(FR, $1, 1)')
+
+dnl KCONTROL_4CHANNELS(reg)
+define(`KCONTROL_4CHANNELS',
+`KCONTROL_CHANNEL(FL, $1, 0),'
+`KCONTROL_CHANNEL(FR, $1, 1),'
+`KCONTROL_CHANNEL(RL, $1, 2),'
+`KCONTROL_CHANNEL(RR, $1, 3)')
+
+dnl KCONTROL_6CHANNELS(reg)
+define(`KCONTROL_6CHANNELS',
+`KCONTROL_CHANNEL(FL, $1, 0),'
+`KCONTROL_CHANNEL(FR, $1, 1),'
+`KCONTROL_CHANNEL(RL, $1, 2),'
+`KCONTROL_CHANNEL(RR, $1, 3),'
+`KCONTROL_CHANNEL(FC, $1, 4),'
+`KCONTROL_CHANNEL(LFE, $1, 5)')
+
+dnl KCONTROL_8CHANNELS(reg)
+define(`KCONTROL_8CHANNELS',
+`KCONTROL_CHANNEL(FL, $1, 0),'
+`KCONTROL_CHANNEL(FR, $1, 1),'
+`KCONTROL_CHANNEL(RL, $1, 2),'
+`KCONTROL_CHANNEL(RR, $1, 3),'
+`KCONTROL_CHANNEL(FC, $1, 4),'
+`KCONTROL_CHANNEL(LFE, $1, 5),'
+`KCONTROL_CHANNEL(SL, $1, 6),'
+`KCONTROL_CHANNEL(SR, $1, 7)')
```

Comment:
This is a bit controversial. It's not because there are 8 channels that they will follow the speaker arrangement that you describe here. It's probably a better idea to keep the names less precise, e.g. vol_ch0..vo_chl7, to avoid confusions.
Take the example of capture on the GP-MRB, the channels are FM, AUX and mic, nothing to do with 7.1 speaker notation

Reply:
I followed the topology definition in alsa lib, else it would be built failed

* The channel map reg is the register offset for the control, shift is the
 * bit shift within the register for the channel and the section name is the
 * channel name and can be one of the following :-
 *
 * <pre>
 *  * mono		# mono stream
 *  * fl 		# front left
 *  * fr		# front right
 *  * rl		# rear left
 *  * rr		# rear right
 *  * fc		# front center
 *  * lfe		# LFE
 *  * sl		# side left
 *  * sr		# side right
 *  * rc		# rear center